### PR TITLE
Check added to PcapNg processing

### PR DESCRIPTION
**Pull request by guedou**
This PR fixes issues discovered by oss-fuzz.

### CODE REVIEW

File: scapy/utils.py [hunk 1]
```diff
@@ -1635,25 +1640,42 @@ def _read_block_shb(self):
...
-        if major != 1:
-            warning(f"SHB Major version {major} unsupported !")
+        try:
+            major = struct.unpack(self.endian + "H", _major)[0]
+        except struct.error:
+            warning("PcapNg: Could not read major value")
             raise EOFError
+        if major != 1:
+            warning(f"PcapNg: SHB Major version {major} unsupported !")
+            raise EOFErro
```
Comment (gpotter2) [inline scapy/utils.py:1666]: typo

**Comment by evverx** [comment]
I apologize for bringing this up here but since it's related to OSS-Fuzz I wonder if it would be OK if I opened https://github.com/google/oss-fuzz/pull/9629 again?

**Comment by guedou** [comment]
@evverx I am fine with the proposed PR that loads all layers. However, I will prefer to keep the reports to the Scapy maintainers only.

**Comment by evverx** [comment]
Fair enough. I fuzz scapy elsewhere so I don't need access to bug reports on OSS-Fuzz (though it would make it easier to filter out duplicates, comment on issues like https://github.com/secdev/scapy/issues/3145#issuecomment-1454702817 reproducible by that fuzz target in certain environments only and so on). I'll open a PR without my email address then.

**Comment by guedou** [comment]
The others maintainers might have a different opinion.

**Comment by gpotter2** [comment]
I'm personally fine with you having your email there.

Two reasons:
- oss-fuzz has currently never reported anything worth worrying about (due to the nature of Scapy, we can't really have anything much more serious than a DoS. It makes those reports not that important).
- You've provided valuable contributions to Scapy, and I believe you're asking in good faith.

**Comment by p-l-** [comment]
@guedou Do you think it would be interesting to add some problematic files as test cases here? Or do we consider oss-fuzz to be enough?

I agree w/ @gpotter2 re. adding @evverx mail.

**Comment by guedou** [comment]
One test case is small (~100 bytes), while the two others one are big (1k+ bytes). I could try to reduce the big ones if you think that it is worth the effort.

**Comment by p-l-** [comment]
Not necessarily, that was more a question than a change request to your PR!

**Review by p-l- — DISMISSED** [review]

**Comment by guedou** [comment]
@gpotter2 I managed to strip down the test cases to 12, 100 and 140 bytes. If that's OK for you, I will add the corresponding unit tests.

**Comment by evverx** [comment]
> Do you think it would be interesting to add some problematic files as test cases here?

FWIW It should be possible to turn on CIFuzz: https://google.github.io/oss-fuzz/getting-started/continuous-integration/. It downloads corpora accumulated by OSS-Fuzz (including test cases that triggered issues in the past) and runs fuzz targets with them. It doesn't catch all the issues but it should be enough to catch most regressions when PRs are opened. Though it looks like OSS-Fuzz is updating the toolchains, interpreters and some other things so I'd wait for them to finish just in case to avoid running into things like https://github.com/google/oss-fuzz/issues/11886.

**Comment by evverx** [comment]
(I opened https://github.com/google/oss-fuzz/pull/11912. It should hopefully make the regression test from https://github.com/secdev/scapy/pull/4378 a bit more useful)

**Review by p-l- — APPROVED** [review]

**Pull request merged**